tools/arbital.py

```
@@ -0,0 +1,106 @@
+import argparse
+import os
+import shutil
+from tqdm import tqdm
+
+from arbital.download import (
+    ARBITAL_SUBSPACES,
+    get_page,
+    get_arbital_page_aliases,
+)
+
+DEFAULT_CACHE_DIR = 'data/cache/arbital'
+DEFAULT_DATA_DIR = 'data/arbital'
+
+
+def write_page(page, path):
+    alias = page['alias']
+    title = page['title']
+    created = page['pageCreatedAt']
+    text = page['text']
+    # TODO: play with formatting here
+    data = f"{title}\n\n{created}\n\n{text}"
+    if not os.path.exists(path):
+        os.makedirs(path, exist_ok=True)
+    fn = os.path.join(path, f'arbital_{alias}.txt')
+    with open(fn, 'w') as f:
+        f.write(data)
+
+def cmd_list(args):
+    aliases = get_arbital_page_aliases(
+        subspace=args.subspace,
+        cache_dir=args.cache_dir,
+    )
+    print(f'{len(aliases)} total pages in subspace `{args.subspace}`.')
+    print(aliases)
+
+def cmd_fetch(args):
+    aliases = get_arbital_page_aliases(
+        subspace=args.subspace,
+        cache_dir=args.cache_dir,
+    )
+    assert args.alias in aliases, 'Article does not exist for this alias'
+    page = get_page(
+        args.alias,
+        subspace=args.subspace,
+        cache_dir=args.cache_dir
+    )
+    write_page(page=page, path=args.path)
+
+def cmd_fetch_all(args):
+    assert args.subspace in ARBITAL_SUBSPACES, 'Not a valid Arbital subspace.'
+    aliases = get_arbital_page_aliases(
+        subspace=args.subspace,
+        cache_dir=args.cache_dir
+    )
+    for alias in tqdm(aliases, disable=not args.progress):
+        page = get_page(alias, args.subspace, args.cache_dir)
+        write_page(page=page, path=args.path)
```

would suggest swallowing errors here
```py
        try:
            page = get_page(alias, args.subspace, args.cache_dir)
            write_page(page=page, path=args.path)
        except JSONDecodeError:
            print(f'Failed to fetch {alias}')
```